sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/aio/_client_async.py

```diff
@@ -897,6 +898,41 @@ async def _message_generator_async(self, timeout=None):
             if self._shutdown:
                 await self.close_async()
 
+    async def _on_disposition_received_async(self, message_delivery, reason, state):
+        for state_key in state:
```

Comment:
does state usually have only 1 value in the list? otherwise we might just exit with the last value

Reply:
can check